app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,24 @@
+class OmniauthCallbacksController < Devise::OmniauthCallbacksController
+
+  def do_omniauth
+    @user = User.from_omniauth(request.env["omniauth.auth"])
+    provider_data = @user.provider
```

Comment:
Optional: This variable isn't being used anywhere else in the code, so you can delete the line.